re/service.go

```diff
@@ -227,3 +345,94 @@ func (re *re) process(ctx context.Context, r Rule, msg *messaging.Message) error
 		return re.pubSub.Publish(ctx, m.Channel, m)
 	}
 }
+
+func (re *re) StartScheduler(ctx context.Context) error {
+	ticker := re.newTicker(time.Minute)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-ticker.C:
+			startDateTime := time.Now().Add(timeZone)
+
+			pm := PageMeta{
+				Status:          EnabledStatus,
+				ScheduledBefore: &startDateTime,
+			}
+
+			page, err := re.repo.ListRules(ctx, pm)
+			if err != nil {
+				re.errors <- err
+				continue
+			}
+
+			for _, rule := range page.Rules {
+				if re.shouldRunRule(rule, startDateTime) {
+					go func(r Rule) {
+						msg := &messaging.Message{
+							Channel: r.InputChannel,
+							Created: startDateTime.Unix(),
+						}
+						re.errors <- re.process(ctx, r, msg)
+					}(rule)
+				}
+			}
+		}
+	}
+}
+
+func (re *re) newTicker(t time.Duration) *time.Ticker {
+	if re.ticker != nil {
+		return re.ticker
+	}
+	return time.NewTicker(t)
+}
```

Comment:
Remove. StartScheduler will be called only once anyway (no need to make it "singleton" in any way).